src/quo/components/tags/tag.cljs

```diff
@@ -47,13 +48,22 @@
                           24 12)
        :color           icon-color}])
    (when (= type :emoji)
-     [text/text
-      {:style {:margin-right 4}
-       :size  (case size
-                32 :paragraph-1
-                24 :paragraph-2
-                nil)}
-      resource])
+     (let [image-emoji? (int? resource)]
```

Comment:
This part is a bit convoluted. What about this:

```clojure
(defn- emoji-comp
  [size resource]
  (let [width (case size
                32 20
                24 12
                nil)]
    (if (string? resource)
      [rn/text {:style {:margin-right 4 :font-size width}}
       resource]
      [rn/image
       {:source resource
        :style  (cond-> {:margin-right 4}
                  (= size 32) (assoc :width width :height 20)
                  (= size 24) (assoc :width width :height 12))}])))

;; Then use it:
(when (= type :emoji)
  [emoji-comp size resource])
```

Changes:

- Break apart components, this allows Reagent to completely skip processing the `emoji-comp` if props other than `size` and `resource` change.
- `cond->` instead of `merge` + `case`.
- `react-native.core/image` instead of `legacy.status-im.ui.components.react`.